examples/adaptive_library/BPMTriangularRotorNotches.py

```diff
@@ -0,0 +1,231 @@
+
+
+# %%
+# Load file if required
+# ---------------------
+# Check if a file is loaded already.
+# If not, open the e9 IPM motor template,
+# save the file to a temporary folder.
+if mc.get_variable("CurrentMotFilePath_MotorLAB") == "":
+    # Disable popup messages
+    mc.set_variable("MessageDisplayState", 2)
+    mc.set_visible(True)
+    mc.load_template("e9")
+    # TODO: Should we keep Motor-CAD open if we have launched a new instance,
+    #  so the user can see what's been done?
+
+    # Open relevant file
+    working_folder = Path(tempfile.gettempdir()) / "adaptive_library"
+    try:
+        shutil.rmtree(working_folder)
+    except:
+        pass
+
+    Path.mkdir(working_folder)
+    mot_name = "BPMTriRotorNotches"
+    mc.save_to_file(working_folder / (mot_name + ".mot"))
+
+    # Disable adaptive templates
+    mc.set_variable("GeometryTemplateType", 0)
+
+# %%
+# Set Adaptive Parameters if required
+# -----------------------------------
+# Four Adaptive Parameters are required
+# for this adaptive template.
+# These are used to define the
+# number of rotor notches to be added,
+# their position and size.
+#
+# If the Adaptive Parameters have already been set in the current Motor-CAD file,
+# their current values will be used.
+# Otherwise, the Adaptive Parameters will be defined
+# and set to default values.
+#
+# The function ``set_default_parameter`` is defined to check if a parameter exists,
+# and if not, create it with a default value.
+
+
+def set_default_parameter(parameter_name, default_value):
+    try:
+        mc.get_adaptive_parameter_value(parameter_name)
+    except pymotorcad.MotorCADError:
+        mc.set_adaptive_parameter_value(parameter_name, default_value)
+
+
+# %%
+# Use the ``set_default_parameter`` to set the required parameters if undefined
+set_default_parameter("Notch Angle", 0)
+set_default_parameter("Notch Sweep", 2)
+set_default_parameter("Notch Depth", 1)
+set_default_parameter("Notches per Pole", 2)
+
+
+# %%
+# Get required parameters and objects
+# -----------------------------------
+# Get the Adaptive Parameters specified in Motor-CAD, and their values
+notch_angle = mc.get_adaptive_parameter_value("notch angle")
+notch_angular_width = mc.get_adaptive_parameter_value("notch sweep")
+notch_depth = mc.get_adaptive_parameter_value("notch depth")
+number_notches = int(mc.get_adaptive_parameter_value("notches per pole"))
+
+# %%
+# Get the standard template rotor region from Motor-CAD.
+# Calculate the rotor radius
+# and define the rotor centre coordinates.
+rotor_region = mc.get_region("Rotor")
+
+rotor_radius = mc.get_variable("RotorDiameter") / 2
+duplication_angle = 360 / rotor_region.duplications
+
+# %%
+# Add an ``if`` statement to account for the case
+# when a notch crosses the lower symmetry boundary.
+# This resets ``notch_angle``
+# to half the ``notch_angular_width``.
+if notch_angle > (duplication_angle / (2 * (number_notches - 1) + 1)) - (notch_angular_width / 2):
```

Comment:
This check should match the reset value, so I think it should be 
if notch_angle > (duplication_angle / (2 * number_notches)) - notch_angular_width / 2: